code/modules/holiday/holidays.dm

```diff
@@ -33,7 +33,7 @@
 	if(!end_month)
 		end_month = begin_month
 	if(begin_week && begin_weekday)
-		if(begin_week == ww && begin_weekday == ddd)
+		if(begin_week == ww && begin_weekday == ddd && begin_month == mm)
```

Comment:
Just curious, why is this nested to begin with?

Also shouldn't 35 check if(begin_month) as well